### ISSUE TITLE
Add support for premium_tier on guilds (once we know what it is)

### ISSUE THREAD

**Issue by queer** [mewna/catnip#316]
Example of a guild object containing this value:
```Javascript
{
  "voice_states" : [],
  "verification_level" : 0,
  "vanity_url_code" : null,
  "unavailable" : false,
  "system_channel_id" : "###",
  "splash" : null,
  "roles" : [],
  // Well would you look at that!
  "premium_tier" : 0,
  "owner_id" : "128316294742147072",
  "name" : "###",
  "mfa_level" : 0,
  "members" : [],
  "member_count" : 11,
  "lazy" : true,
  "large" : false,
  "joined_at" : "###",
  "id" : "###",
  "icon" : null,
  "features" : [],
  "explicit_content_filter" : 0,
  "emojis" : [],
  "description" : null,
  "default_message_notifications" : 0,
  "channels" : [],
  "banner" : null,
  "application_id" : null,
  "afk_timeout" : 300,
  "afk_channel_id" : null
}
```
Note the `premium_tier` field above `owner_id`.

**Comment by queer** [comment]
Note that this does NOT mean "immediately open a PR," but rather "we should PR this once we know what it's for / that it won't be removed"

**Comment by queer** [comment]
Closed in favour of #329
